src/basepaths.rs

```diff
@@ -481,6 +483,21 @@ mod tests {
         let helper = testhelper::TestHelper::new()?.with_git_repo()?;
         let modified = helper.modify_files()?;
 
+        let hook = r#"
+                !#/bin/bash
+                echo "X"
+            "#;
+
+        let mut file_path = helper.root();
+        file_path.push(".git/hooks/post-checkout");
+        helper.write_file(&file_path, hook)?;
+
+        let file = File::open(&file_path.into_os_string())?;
+        let metadata = file.metadata()?;
+        let mut permissions = metadata.permissions();
+        permissions.set_mode(0o755);
+        assert_eq!(permissions.mode(), 0o755);
```

Comment:
Despite this I'm seeing:

```
failures:

---- basepaths::tests::git_staged_mode_with_changes stdout ----
Error: Got unexpected stderr output from `git stash --keep-index`:
hint: The '.git/hooks/post-checkout' hook was ignored because it's not set as executable.
hint: You can disable this warning with `git config advice.ignoredHook false`.
```

So, I'm not sure if this check is misleading or there's some filesystem issue with the temp dir that's getting in the way.

Reply:
My usual trick is to disable deleting the temp dir so I can examine it after tests fail.

Reply:
Oh, I see the problem. Calling `set_permissions` doesn't change anything on the filesystem. You need to call `fs::set_permissions` - see https://doc.rust-lang.org/std/fs/fn.set_permissions.html

Reply:
Thanks! I'll pick this up again on Monday. I appreciate all the guidance.